.github/workflows/main.yml

```diff
@@ -38,6 +38,12 @@ jobs:
           fetch-tags: true
           ref: ${{ github.event.pull_request && github.event.pull_request.head.sha || github.ref }}
 
+      - name: Get CPU core count
+        id: core_count
+        run: |
+          CORES=$(nproc --all)
```

Comment:
* shell expansion should work everywhere (at lerast for this simple case)
* I think `--all` is wrong. w/o it, nproc gives the number of usable cores (i.e., including system restrictions), which is what we should use. The number of hardware cores does not matter if the task affinity (or similar) disallows their use... (note: currently they are equal)